From d3730e7937ed888db6437d89eccc6bf15daa6e15 Mon Sep 17 00:00:00 2001
From: Gustavo Padovan <gustavo.padovan@collabora.co.uk>
Date: Tue, 31 May 2016 16:59:09 -0300
Subject: [PATCH 57/88] staging/android: remove 'destroyed' member from struct
 sync_timeline
Git-commit: 342952d3a5c4d128649c7fd8e34d7d5eb6d036aa
Patch-mainline: v4.8-rc1
References: fate#321223

'destroyed' was set but not used ny anyone.

Signed-off-by: Gustavo Padovan <gustavo.padovan@collabora.co.uk>
Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Reviewed-by: Sumit Semwal <sumit.semwal@linaro.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/staging/android/sw_sync.c | 5 -----
 drivers/staging/android/sync.h    | 5 +----
 2 files changed, 1 insertion(+), 9 deletions(-)

diff --git a/drivers/staging/android/sw_sync.c b/drivers/staging/android/sw_sync.c
index 72262ba4df7f..1f956b99b880 100644
--- a/drivers/staging/android/sw_sync.c
+++ b/drivers/staging/android/sw_sync.c
@@ -107,11 +107,6 @@ static void sync_timeline_put(struct sync_timeline *obj)
  */
 static void sync_timeline_destroy(struct sync_timeline *obj)
 {
-	obj->destroyed = true;
-	/*
-	 * Ensure timeline is marked as destroyed before
-	 * changing timeline's fences status.
-	 */
 	smp_wmb();
 
 	sync_timeline_put(obj);
diff --git a/drivers/staging/android/sync.h b/drivers/staging/android/sync.h
index 3a508515f12b..3c551f5401ee 100644
--- a/drivers/staging/android/sync.h
+++ b/drivers/staging/android/sync.h
@@ -25,10 +25,8 @@
  * @kref:		reference count on fence.
  * @drv_name:		drv_name of the driver using the sync_timeline
  * @name:		name of the sync_timeline. Useful for debugging
- * @destroyed:		set when sync_timeline is destroyed
  * @child_list_head:	list of children sync_pts for this sync_timeline
- * @child_list_lock:	lock protecting @child_list_head, destroyed, and
- *			fence.status
+ * @child_list_lock:	lock protecting @child_list_head and fence.status
  * @active_list_head:	list of active (unsignaled/errored) sync_pts
  * @sync_timeline_list:	membership in global sync_timeline_list
  */
@@ -38,7 +36,6 @@ struct sync_timeline {
 	char			name[32];
 
 	/* protected by child_list_lock */
-	bool			destroyed;
 	int			context, value;
 
 	struct list_head	child_list_head;
-- 
2.12.0
